flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java

```diff
@@ -909,47 +694,172 @@ void testDropMaterializedTable(@InjectClusterClient RestClusterClient<?> restClu
+                service.refreshMaterializedTable(
+                        sessionHandle,
+                        objectIdentifier.asSerializableString(),
+                        true,
+                        null,
+                        Collections.emptyMap(),
+                        Collections.emptyMap(),
+                        Collections.emptyMap());
+        assertThatThrownBy(
+                        () ->
+                                awaitOperationTermination(
+                                        service, sessionHandle, invalidRefreshTableHandle1))
+                .rootCause()
+                .isInstanceOf(ValidationException.class)
+                .hasMessage(
+                        String.format(
+                                "Scheduler time not properly set for periodic refresh of table %s",
+                                ObjectIdentifier.of(
+                                                fileSystemCatalogName,
+                                                TEST_DEFAULT_DATABASE,
+                                                "my_materialized_table")
+                                        .asSerializableString()));
+
+        // refresh the materialized table with invalid schedule time
+        String invalidTime = "20240103 00:00:00.000";
+        OperationHandle invalidRefreshTableHandle2 =
+                service.refreshMaterializedTable(
+                        sessionHandle,
+                        objectIdentifier.asSerializableString(),
+                        true,
+                        invalidTime,
+                        Collections.emptyMap(),
+                        Collections.emptyMap(),
+                        Collections.emptyMap());
+
+        assertThatThrownBy(
+                        () ->
+                                awaitOperationTermination(
+                                        service, sessionHandle, invalidRefreshTableHandle2))
+                .rootCause()
+                .isInstanceOf(SqlExecutionException.class)
+                .hasMessage(
+                        String.format(
+                                "Failed to parse a valid partition value for the field 'ds' in table %s using the scheduler time '20240103 00:00:00.000' based on the date format 'yyyy-MM-dd HH:mm:ss'.",
+                                ObjectIdentifier.of(
+                                                fileSystemCatalogName,
+                                                TEST_DEFAULT_DATABASE,
+                                                "my_materialized_table")
+                                        .asSerializableString()));
     }
 
-    private List<RowData> fetchTableData(SessionHandle sessionHandle, String query) {
-        OperationHandle queryHandle =
-                service.executeStatement(sessionHandle, query, -1, new Configuration());
+    private void removePartitionValue(List<Row> data, String partition) {
```

Comment:
This method name looks a little strange, you just remove one record in the partition, but name it `removePartitionValue`. My core point is that: why do you need this method? Does this have other way to implement your test purpose?

Reply:
This has already been removed and replaced with adding data corresponding to the partition.